### PR TITLE
Convert remaining packages which necessitate 1es-templates/esrp release

### PR DESCRIPTION
**Pull request by scbedd**
The following yml are associated with projects that need to publish to npmjs.org. This PR simply removes each's individual `release` stage in favor of a single added stage. After walking each package within `azure-sdk-tools`, I believe these are the only remaining builds that must become 1es compliant.

  - [x] ~~tools/sdk-generation-pipeline/ci.yml~~ Deleted in `main`
  - [x] tools/oav-traffic-converter/ci.yml
  - [x] tools/mock-service-host/ci.yml
  - [x] tools/js-sdk-release-tools/ci.yml
  - [x] ~~tools/apiview/parsers/js-api-parser/ci.yml~~ Only publishes to internal feed
  - [x] tools/apiview/emitters/typespec-apiview/ci.yml -- this only needs ESRP Release stage, already 1es compliant.
  - [x] ~~eng/pipelines/templates/stages/archetype-autorest-preview.yml~~ This yml is referenced _nowhere_ that I can tell. Not from another template and not direct from a build definition. 
  - [x] tools/tsp-client/ci.yml  -- this only needs ESRP Release stage, already 1es

**FOR REVIEWERS** I advise disabling whitespace diffs it massively simplifies looking at what this PR does. Much more re-use than code deletion.

Related to #7941

### CODE REVIEW

**Comment by scbedd** [comment]
/check-enforcer override

The `tsp-client` failure is merely because I cancelled the release itself.

File: eng/scripts/determine-js-release-tag.ps1 [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Read the package.json file
+$packageJsonPath = "$PSScriptRoot/../../package.json"
+$packageJson = Get-Content $packageJsonPath -Raw | ConvertFrom-Json
+
+# Function to check if a version is non-GA
```
Comment (weshaggard) [inline eng/scripts/determine-js-release-tag.ps1:5]: Is it worth using our SemVer.ps1 here? Do we have non-standard prerelease tags?

File: eng/pipelines/templates/stages/archetype-sdk-publish-js.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    - pwsh: |
+                        Write-Host "Will deploy with tag of $(Tag)"
+                        Get-ChildItem "$(Pipeline.Workspace)/drop" -Recurse -Force `
+                          | Where-Object { $_.Name -like "*.tgz" } `
+                          | Copy-Item -Destination "$(Build.ArtifactStagingDirectory)"
+
+                        Get-ChildItem "$(Build.ArtifactStagingDirectory)" -Recurse -Force | % { Write-Host $_.FullName }
+                      displayName: Move artifact to $(Build.ArtifactStagingDirectory)
+
+                    - task: EsrpRelease@7
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-sdk-publish-js.yml:58]: Thank you for cleaning these up and converting them to ESRP. FYI @ckairen as we will eventually want common publish template here that can be used outside of the tools repo as well.

**Review by weshaggard — APPROVED** [review]
Looks reasonable assuming test pipelines work.

File: eng/scripts/determine-js-release-tag.ps1 (hunk 1 above)
Comment (scbedd) [reply eng/scripts/determine-js-release-tag.ps1:5]: I'll take a look, thanks for the suggestion 👍

Comment (scbedd) [reply eng/scripts/determine-js-release-tag.ps1:5]: Yes, I'll parse the version from package.json, then set `beta` if isPrerelease` is true`.

**Comment by weshaggard** [comment]
Ray recently deleted `tools/sdk-generation-pipeline/ci.yml` so you can remove those changes.

**Comment by scbedd** [comment]
Will do so. Needed to get in here to make the SemVer changes anyway.

**Comment by scbedd** [comment]
My claim is that

1. The existing build failures are already existing. Going to ignore them.
2. For the failed `SDL Sources`, I believe this is only on [the merge PR](https://dev.azure.com/azure-sdk/public/_build/results?buildId=3890568&view=results) causing some weird interaction with the submodules. This doesn't occur when just run straight from the branch. And the branch is updated from `main` as of this morning.

Just going to merge this and then deal with the fallout.

File: eng/pipelines/templates/stages/archetype-sdk-publish-js.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    steps:
+                    - checkout: self
+
+                    - download: current
+                      artifact: ${{ parameters.ArtifactName }}
+                      timeoutInMinutes: 5
+
+                    - task: PowerShell@2
+                      inputs:
+                        filePath: '$(Build.SourcesDirectory)/eng/scripts/determine-release-tag.ps1'
```
Comment (catalinaperalta) [inline eng/pipelines/templates/stages/archetype-sdk-publish-js.yml:45]: This file doesnt match the new one being added in the PR. Seems like the file name should be `determine-js-release-tag.ps1`

**Pull request merged**